noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/ui/screens/LoginScreen.kt

```diff
@@ -184,4 +180,13 @@ fun LoginScreen(navController: NavHostController, loginViewModel: LoginViewModel
             }
         }
     }
+
+    LaunchedEffect(viewState?.isSuccess) {
+        if (viewState?.isSuccess == true) {
+            navController.navigate(Screen.Notes.route) {
+                launchSingleTop = true
+                popUpTo(NOTY_NAV_HOST_ROUTE)
+            }
+        }
+    }
```

Comment:
Calling navigation methods directly from a `@Composable` function is not good and should be strictly avoided. It should be part of a callback. Using `LaunchedEffect` helps in achieving the same and finally, it fixes the navigation issues 😎